django_x509/base/models.py

```diff
@@ -291,6 +291,8 @@ def _generate(self):
         generates a new x509 certificate (CA or end-entity)
         """
         key = crypto.PKey()
+        if not self.key_length:
+            return
```

Comment:
We should update the fields instead and make them required and non-nullable. What do you think @nemesisdesign ?

https://github.com/openwisp/django-x509/blob/16fa68f783a2b45f788a472e1ba573a0f061c781/django_x509/base/models.py#L114-L129

Reply:
I agree with @pandafy.